bigquery_etl/query_scheduling/task.py

```diff
@@ -75,12 +76,36 @@ def of_query(cls, query_file):
         metadata = Metadata.of_sql_file(query_file)
         return cls(query_file, metadata)
 
-    def get_dependencies(self):
+    def _get_referenced_tables(self, client):
```

Comment:
We might want to memoize this using a `functools.cached_property`decorator. If this has to be invoked multiple times, it could slow things down.